src/shell/shell.cpp

```diff
@@ -736,6 +754,98 @@ class AUTOEXEC final : public Module_base {
 	}
 };
 
+// Specify a 'Drive' config object with allowed key and value types
+static Config specify_drive_config()
+{
+	auto conf = Config();
+
+	// Define the [drive] section
+	constexpr auto changeable_at_runtime = false;
+	auto prop = conf.AddSection_prop("drive", nullptr, changeable_at_runtime);
+
+	// Define the allowed keys and types
+	constexpr auto on_startup = Property::Changeable::OnlyAtStart;
+	const char *drive_types[] = {"dir", "floppy", "cdrom", "overlay", nullptr};
+	(void)prop->Add_string("type", on_startup, "")->Set_values(drive_types);
+	(void)prop->Add_string("label", on_startup, "");
+	(void)prop->Add_string("path", on_startup, "");
+	(void)prop->Add_string("override_drive", on_startup, "");
+	(void)prop->Add_bool("verbose", on_startup, true);
+
+	return conf;
+}
+
+// Parse a 'Drive' config file and return object with allowed key and value types
+static std::tuple<std::string, std::string, std::string> parse_drive_conf(
+        std::string drive_letter, const std_fs::path &conf_path)
+{
+	// Default return values
+	constexpr auto default_args = "";
+	constexpr auto default_path = "";
+
+	// If the conf path doesn't exist, at least return the default quiet arg
+	if (!path_exists(conf_path))
+		return {drive_letter, default_args, default_path};
+
+	// If we couldn't parse it, return the defaults
+	auto conf = specify_drive_config();
+	if (!conf.ParseConfigFile("drive", conf_path.string()))
+		return {drive_letter, default_args, default_path};
+
+	const auto settings = static_cast<Section_prop *>(conf.GetSection("drive"));
+
+	// Construct the mount arguments
+	const auto override_drive = std::string(settings->Get_string("override_drive"));
+	if (override_drive.length() == 1 && override_drive[0] >= 'a' && override_drive[0] <= 'y')
+		drive_letter = override_drive;
+	else if (override_drive.length()) {
+		LOG_ERR("AUTOMOUNT: %s: setting 'override_drive = %s' is invalid", conf_path.string().c_str(), override_drive.c_str());
+		LOG_ERR("AUTOMOUNT: The override_drive setting can be left empty or a drive letter from 'a' to 'y'");
+	}
+
+	auto drive_type = std::string(settings->Get_string("type"));
```

Comment:
`std::string drive_type = settings->Get_string("type")` wouldn't work, would it?

Reply:
const?

Reply:
Thanks! Done.

I do need to write into it if it's present, to constructing the `-t <type>` chunk of the mount command-line arguments:

Here's the update:

``` c++
std::string drive_type = settings->Get_string("type");
if (drive_type.length())
	drive_type.insert(0, " -t ");
```